src/script/utils/url.ts

```diff
@@ -11,7 +11,7 @@ export function isUrl(url: string): boolean {
 }
 
 export function resolveUrl(baseUrl: string, url: string): string | undefined {
-  let parsedUrl = undefined;
+  let parsedUrl: URL | undefined = undefined;
```

Comment:
due to these changes, let's hold on merging this until yours are merged. I saw some changes in one of your PRs and I like your changes better tbh